### PR TITLE
Range-based auto-firing works

### PR DESCRIPTION
**Pull request by BenWertz**
We tested it, we fixed the bug (it was just the trapezoid profile restarting because we called setHoodPosition() in execute()), and it works. It fires _really_ well, actually a bit better than the preset distance buttons since it doesn't need to be at an exact distance.

### CODE REVIEW

**Review by roboticsgeek — APPROVED** [review]
Verified works /w drive team check. Big improvement.

File: src/main/java/frc/robot/commands/ShooterTargetCommand.java [hunk 1]
```diff
@@ -18,105 +18,78 @@
...
 
-  HubTracking hubTracking;
-  Shooter shooter;
-  HubData data;
-  double range = 0;
-  boolean dataCollected = false;
-
-  InterpolatorTable flywheelTable = new InterpolatorTable(
+  
+  public static InterpolatorTable flywheelTable = new InterpolatorTable(
```
Comment (r-wallis) [inline src/main/java/frc/robot/commands/ShooterTargetCommand.java:22]: one thing I don't like about this approach is that the interpolator table has to exist in 2 places now ... a longer term solution to that would be great
Then again, I don't see this in the other command, so ignore this!

File: src/main/java/frc/robot/commands/ShooterRangeTargetCommand.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    hubTracking=hubTracking_;
+    data=new HubData();
+    addRequirements(shooter);
+    // Use addRequirements() here to declare subsystem dependencies.
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    SmartDashboard.putBoolean("VISIBLE", hubTracking.isHubVisible());
```
Comment (r-wallis) [inline src/main/java/frc/robot/commands/ShooterRangeTargetCommand.java:31]: I think we need to specify that these are for the hub - so HUB IS VISIBLE ... is that useful in the command though?  Seems like the subsystem should set this.

**Review by r-wallis — APPROVED** [review]

**Comment by BenWertz** [comment]
Okay, merging this now.

**Pull request merged**